### PR TITLE
adds link into SEO Page

### PR DESCRIPTION
**Pull request by cmizzi**
This pull request will allow you to add tags like : 

```html
// HTML Render
<link rel="prev" href="http://example.com/2" />
<link rel="next" href="http://example.com/4" />
```

```php
// PHP
$seo->addLink('prev', array('href' => 'http://example.com/2'));
$seo->addLink('next', array('href' => 'http://example.com/4'));
```

### CODE REVIEW

File: Seo/SeoPage.php [hunk 1]
```diff
@@ -34,11 +34,6 @@ class SeoPage implements SeoPageInterface
     /**
      * @var string
      */
-    protected $linkCanonical;
```
Comment (OskarStark) [inline Seo/SeoPage.php]: This is a BC break

File: Seo/SeoPage.php [hunk 2]
```diff
@@ -49,12 +44,12 @@ class SeoPage implements SeoPageInterface
     /**
      * @var array
      */
-    protected $langAlternates;
+    protected $oembedLinks;
```
Comment (OskarStark) [inline Seo/SeoPage.php]: This is a BC break

File: Seo/SeoPage.php [hunk 3]
```diff
@@ -49,12 +44,12 @@ class SeoPage implements SeoPageInterface
...
 
     /**
      * @var array
      */
-    protected $oembedLinks;
+    protected $links;
```
Comment (OskarStark) [inline Seo/SeoPage.php]: This is a BC break

File: Seo/SeoPage.php [hunk 4]
```diff
@@ -297,7 +291,7 @@ public function hasHeadAttribute($name)
      */
     public function setLinkCanonical($link)
     {
-        $this->linkCanonical = $link;
+        $this->setLink("canonical", array("href" => $link));
```
Comment (OskarStark) [inline Seo/SeoPage.php]: please use ``'`` insteadof ``"``

File: Seo/SeoPageInterface.php [hunk 5]
```diff
@@ -153,4 +153,61 @@ public function addOEmbedLink($title, $link);
...
+     */
+    public function getLinks();
+
+    /**
+     * Remove all link based on the given type
+     *
+     * @param  string $type
+     * @return boolean
+     */
+    public function removeLink($type);
```
Comment (OskarStark) [inline Seo/SeoPageInterface.php]: Adding methods to an existing Interface is BC

**Comment by OskarStark** [comment]
Looks like we can merge this PR in the next major.

Please add an UPGRADE notice or try to handle the BC breaks.
Don't forget the documentation, otherwise this feature can't be merged.

Thank you @cmizzi 

ping @Soullivaneuh

**Comment by OskarStark** [comment]
@cmizzi please make travis happy and fix the CS issues, thank you

**Comment by cmizzi** [comment]
I've fix BC and all Travis CI errors but there's somes errors from files that I didn't update... I'm not able to make travis happy with some external pull request error.

Still some BC fixes needed ?

**Comment by core23** [comment]
Please rebase, squash and run cs fixer @cmizzi 

Please have a look for @OskarStark comments. Removing existing stuff is bad for BC.

**Comment by cmizzi** [comment]
I don't know what happened but, there's all others commit in my pull request... Sorry for that, and now, there's a branch conflict error.

**Pull request closed without merging**